### ISSUE TITLE
Split 'Administer site configuration' into more specific permissions

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#3761]
**Description of the need**
I'd like to give my client access to change the site logo, slogan, etc., but if I give them the 'Administer site configuration' permission, they also have access to a whole bunch of additional pages/settings they don't need to see. I think 'Administer site configuration' is too broad, and we should replace it with more specific permissions.

**Proposed solution**
Have more specific permissions for accessing the different pages under the Configuration admin menu item, to allow for more granular control.

**Alternatives that have been considered**
The [Custom Permissions](https://www.drupal.org/project/config_perms) module for Drupal seems to have evolved out of a similar frustration with this issue, however it provides a UI for creating additional permissions and assigning them to paths. Whereas I'm suggesting extra permissions simply be built into core (no extra UI needed).

**Comment by klonos** [comment]
Yes! I'd be in favor of more granular permissions instead of this "umbrella" permission for everything. I believe that a new `administer basic site information` permission would be a good start.

A UI that allows specifying custom permissions (either based on paths or menu routes) sounds like it should stay in contrib.

**Comment by klonos** [comment]
...reminds me of how great we did back when we introduced a permissions specifically for allowing people to assign user roles (#313):

![image](https://user-images.githubusercontent.com/2423362/151250138-1fc30913-bdf9-4609-bf58-5749a1ebcef7.png)

This is another issue with Drupal that keeps coming up with customers at work, which we don't have to worry about in Backdrop!
